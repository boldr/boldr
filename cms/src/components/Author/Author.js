/* @flow */
import React from 'react';
import { Divider, Card, Comment } from 'semantic-ui-react';
import { Heading } from 'components/index';
import Avatar from '../Avatar';

const Author = (props: { display_name: String, avatar_url: String }) => {
  return (
    <div className="post__sidebar-author">
    <div className="post__sidebar-card">
    <Card.Content>
      <Avatar src={ props.avatar_url } />
      <Card.Header> { props.display_name }</Card.Header>
      </Card.Content>
      </div>
    </div>
  );
};

export default Author;
