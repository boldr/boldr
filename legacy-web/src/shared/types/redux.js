/* @flow */
import type { Store as ReduxStore } from 'redux';

export type ApiAction<T> = {
  key: string,
  payload: T,
};

export type ActionType =
  | 'LOGIN_REQUEST'
  | 'LOGIN_SUCCESS'
  | 'LOGIN_FAILURE'
  | 'LOGOUT_USER'
  | 'LOGOUT_USER_FAIL'
  | 'CHECK_AUTH_REQUEST'
  | 'CHECK_AUTH_SUCCESS'
  | 'CHECK_AUTH_FAILURE'
  | 'SIGNUP_USER_REQUEST'
  | 'SIGNUP_USER_SUCCESS'
  | 'SIGNUP_USER_FAILURE'
  | 'FORGOT_PASSWORD_REQUEST'
  | 'FORGOT_PASSWORD_SUCCESS'
  | 'FORGOT_PASSWORD_FAILURE'
  | 'RESET_PASSWORD_REQUEST'
  | 'RESET_PASSWORD_SUCCESS'
  | 'RESET_PASSWORD_FAILURE'
  | 'LOAD_POSTS_REQUEST'
  | 'LOAD_POSTS_SUCCESS'
  | 'LOAD_POSTS_FAILURE'
  | 'FETCH_ARTICLE_REQUEST'
  | 'FETCH_ARTICLE_SUCCESS'
  | 'FETCH_ARTICLE_FAILURE'
  | 'UPDATE_ARTICLE_REQUEST'
  | 'UPDATE_ARTICLE_SUCCESS'
  | 'UPDATE_ARTICLE_FAILURE'
  | 'SELECT_ARTICLE'
  | 'SELECT_ARTICLE_SUCCESS'
  | 'SELECT_ARTICLE_FAILURE'
  | 'CREATE_ARTICLE_REQUEST'
  | 'CREATE_ARTICLE_SUCCESS'
  | 'CREATE_ARTICLE_FAILURE'
  | 'DELETE_ARTICLE_FAILURE'
  | 'DELETE_ARTICLE_REQUEST'
  | 'DELETE_ARTICLE_SUCCESS';

export type ArticlesReducer = Object & {
  all: Object,
  ids: Array<Object>,
  isFetching: boolean,
  currentArticle: Object,
};

export type BlogReducer = {
  articles: ArticlesReducer,
  tags: any,
};

export type UserInfo = {
  [userId: string]: {
    readyStatus: string,
    err: any,
    info: Object,
  },
};

export type Reducer = {
  routing: any,
  blog: BlogReducer,
};

export type Action =
  | { type: '@boldr/blog/FETCH_ARTICLES_REQUEST' }
  | { type: '@boldr/blog/FETCH_ARTICLES_SUCCESS', payload: NormalizrPayload }
  | { type: '@boldr/blog/FETCH_ARTICLES_FAILURE', err: any }
  | { type: '@boldr/blog/FETCH_ARTICLE_REQUEST', slug: string }
  | {
      type: '@boldr/blog/FETCH_ARTICLE_SUCCESS',
      slug: string,
      payload: Article,
    }
  | { type: '@boldr/blog/FETCH_ARTICLE_FAILURE', slug: string, err: any };

export type Store = ReduxStore<Reducer, Action>;
// eslint-disable-next-line no-use-before-define
export type Dispatch = (
  action: Action | ThunkAction | PromiseAction | Array<Action>,
) => any;
export type GetState = () => Object;
export type ThunkAction = (dispatch: Dispatch, getState: GetState) => any;
export type PromiseAction = Promise<Action>;
