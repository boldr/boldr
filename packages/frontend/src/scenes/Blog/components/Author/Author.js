/* @flow */
import React from 'react';
import styled from 'styled-components';
import classnames from 'classnames';
import Link from 'react-router-dom/Link';
import Avatar from '@boldr/ui/Avatar';
import Heading from '@boldr/ui/Heading';
import { StyleClasses } from '@boldr/ui/theme/styleClasses';

const BASE_ELEMENT = StyleClasses.ARTICLE_SIDEBAR_AUTHOR;

const AvatarName = styled.div`
  display: inline-block;
  vertical-align: middle;
  &:first-child {
    margin-right: 15px;
  }
`;

const BioBlock = styled.div`
  vertical-align: middle;
  margin-bottom: 15px;
  padding-top: 10px;
`;

export type Props = {
  className: string,
  author: Object,
};

const Author = (props: Props) => {
  const { author } = props;
  const classes = classnames(BASE_ELEMENT, props.className);

  return (
    <div className={classes}>
      <AvatarName>
        <Avatar src={author.profile.avatarUrl} role="presentation" />
      </AvatarName>
      <AvatarName>
        <Link to={`/profiles/${author.profile.username}`}>
          <Heading type="h3" text={author.profile.username} />
        </Link>
      </AvatarName>
      <BioBlock>{author.profile.bio}</BioBlock>
    </div>
  );
};
const defaultProps = {
  author: {
    username: 'Placeholder',
  },
};
Author.defaultProps = defaultProps;
export default Author;
