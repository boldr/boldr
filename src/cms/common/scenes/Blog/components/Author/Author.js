/* @flow */
import React from 'react';
import { Card, Feed } from 'semantic-ui-react';

const Author = (props: { display_name: String, avatar_url: String }) => {
  return (
    <div className="post__sidebar-author">
    <Card className="post__sidebar-card">
      <Card.Content>
        <Feed>
          <Feed.Event>
            <Feed.Label image={ props.avatar_url } />
            <Feed.Content>
              <Feed.Summary>
                { props.display_name }
              </Feed.Summary>
            </Feed.Content>
          </Feed.Event>
        </Feed>
      </Card.Content>
    </Card>
      </div>
  );
};

export default Author;
