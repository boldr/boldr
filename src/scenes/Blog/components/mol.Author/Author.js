import React from 'react';
import Avatar from '../../../../components/md/Avatars';
import { Heading } from '../../../../components';

const styles = {
  wrapper: {
    display: 'flex',
    flexWrap: 'wrap',
    alignItems: 'center'
  }
};
const Author = (props) => {
  return (
    <div style={ styles.wrapper }>
      <Avatar src={ props.avatarUrl } />
      <Heading size={ 3 }> { props.displayName }</Heading>
    </div>
  );
};

Author.propTypes = {
  displayName: React.PropTypes.string.isRequired,
  avatarUrl: React.PropTypes.string
};

export default Author;
