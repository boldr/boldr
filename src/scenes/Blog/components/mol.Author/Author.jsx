import React from 'react';

const Author = (props) => {
  return (
    <div>
      { props.user.displayName }
    </div>
  );
};

Author.propTypes = {
  displayName: React.PropTypes.string.isRequired
};

export default Author;
