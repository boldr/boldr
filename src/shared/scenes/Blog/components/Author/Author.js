/* @flow */
import React from 'react';
import styled from 'styled-components';
import classnames from 'classnames';
import Link from 'react-router-dom/Link';
import {Avatar, Heading, Col, Social, StyleClasses} from 'boldr-ui';

const BASE_ELEMENT = StyleClasses.POST_SIDEBAR_AUTHOR;

const AvatarName = styled.div`
  display: inline-block;
  vertical-align: middle;
  &:first-child {
    margin-right: 15px;
  }
`;
const BioBlock = styled.div`
  vertical-align: middle;
  margin-bottom: 15px;
  padding-top: 10px;
`;

type Props = {
  className: string,
  username: string,
  avatarUrl: string,
  lastName: string,
  firstName: string,
  bio: string,
  social: Object,
};
const Author = (props: Props) => {
  const classes = classnames(BASE_ELEMENT, props.className);
  const authorName = (
    <Link to={`/profiles/${props.username}`}>{props.username}</Link>
  );
  return (
    <div className={classes}>
      <AvatarName>
        <Avatar src={props.avatarUrl} role="presentation" />
      </AvatarName>
      <AvatarName><Heading size={3}>{authorName}</Heading></AvatarName>
      <BioBlock>{props.bio}</BioBlock>
      <Social
        facebook
        fburl={props.social.facebook.url}
        twitter
        turl={props.social.twitter.url}
        google
        gurl={props.social.google.url}
        github
        ghurl={props.social.github.url}
        linkedin
        lurl={props.social.linkedin.url}
      />
    </div>
  );
};

Author.defaultProps = {
  avatarUrl: '',
};

export default Author;
