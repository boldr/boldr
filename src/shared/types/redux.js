/* @flow */
import type { Store as ReduxStore } from 'redux';

export type ApiAction<T> = {
  key: string,
  payload: T,
};

export type ActionType =
  | 'LOGIN_REQUEST'
  | 'LOGIN_SUCCESS'
  | 'LOGIN_FAILURE'
  | 'LOGOUT_USER'
  | 'LOGOUT_USER_FAIL'
  | 'CHECK_AUTH_REQUEST'
  | 'CHECK_AUTH_SUCCESS'
  | 'CHECK_AUTH_FAILURE'
  | 'SIGNUP_USER_REQUEST'
  | 'SIGNUP_USER_SUCCESS'
  | 'SIGNUP_USER_FAILURE'
  | 'FORGOT_PASSWORD_REQUEST'
  | 'FORGOT_PASSWORD_SUCCESS'
  | 'FORGOT_PASSWORD_FAILURE'
  | 'RESET_PASSWORD_REQUEST'
  | 'RESET_PASSWORD_SUCCESS'
  | 'RESET_PASSWORD_FAILURE'
  | 'LOAD_POSTS_REQUEST'
  | 'LOAD_POSTS_SUCCESS'
  | 'LOAD_POSTS_FAILURE'
  | 'FETCH_POST_REQUEST'
  | 'FETCH_POST_SUCCESS'
  | 'FETCH_POST_FAILURE'
  | 'UPDATE_POST_REQUEST'
  | 'UPDATE_POST_SUCCESS'
  | 'UPDATE_POST_FAILURE'
  | 'SELECT_POST'
  | 'SELECT_POST_SUCCESS'
  | 'SELECT_POST_FAILURE'
  | 'CREATE_POST_REQUEST'
  | 'CREATE_POST_SUCCESS'
  | 'CREATE_POST_FAILURE'
  | 'DELETE_POST_FAILURE'
  | 'DELETE_POST_REQUEST'
  | 'DELETE_POST_SUCCESS';

export type PostsReducer = {
  all: Object,
  ids: Array<Object>,
  isFetching: boolean,
  currentPost: Object,
};

export type BlogReducer = {
  posts: PostsReducer,
  tags: any,
  comments: any,
};

export type UserInfo = {
  [userId: string]: {
    readyStatus: string,
    err: any,
    info: Object,
  },
};

export type Reducer = {
  routing: any,
  blog: BlogReducer,
};


export type Action =
  { type: '@boldr/blog/FETCH_POSTS_REQUEST' } |
  { type: '@boldr/blog/FETCH_POSTS_SUCCESS', data: Array<Object> } |
  { type: '@boldr/blog/FETCH_POSTS_FAILURE', err: any } |
  { type: '@boldr/blog/FETCH_POST_REQUEST', userId: string } |
  { type: '@boldr/blog/FETCH_POST_SUCCESS', userId: string, data: Object } |
  { type: '@boldr/blog/FETCH_POST_FAILURE', userId: string, err: any };

export type Store = ReduxStore<Reducer, Action>;
// eslint-disable-next-line no-use-before-define
export type Dispatch = (action: Action | ThunkAction | PromiseAction | Array<Action>) => any;
export type GetState = () => Object;
export type ThunkAction = (dispatch: Dispatch, getState: GetState) => any;
export type PromiseAction = Promise<Action>;
