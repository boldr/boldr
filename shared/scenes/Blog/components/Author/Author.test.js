import React from 'react';
import { createStore } from 'redux';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Author from './Author';

describe('<Author />', () => {
  const wrapper = shallow(<Author avatar_url={ 'abcd' } display_name="hey" />);
  it('renders <Author /> without breaking', () => {
    expect(wrapper.find('div').length).toBe(1);
    expect(wrapper.instance().props.display_name).toBe('hey');
  });
  it('renders snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
