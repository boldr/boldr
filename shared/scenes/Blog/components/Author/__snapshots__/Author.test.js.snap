exports[`<Author /> renders snapshot 1`] = `
<div
  className="boldr-post__sidebar-author">
  <List>
    <ListItem
      activeClassName="md-text--theme-primary"
      component="div"
      disabled={true}
      expanderIconChildren="keyboard_arrow_down"
      leftAvatar={
        <Avatar
          role="presentation"
          src="abcd"
          suffixes={
            Array [
              "red",
              "pink",
              "purple",
              "deep-purple",
              "indigo",
              "blue",
              "light-blue",
              "cyan",
              "teal",
              "green",
              "light-green",
              "lime",
              "yellow",
              "amber",
              "orange",
              "deep-orange",
              "brown",
              "grey",
              "blue-grey",
            ]
          } />
      }
      primaryText={NaN} />
  </List>
</div>
`;
